plugin/trino-plugin-archetype/src/main/resources/archetype-resources/src/main/java/__classPrefix__Module.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package $package;
+
+import com.google.inject.Binder;
+import com.google.inject.Module;
+import com.google.inject.Scopes;
+import io.trino.spi.NodeManager;
+import io.trino.spi.type.TypeManager;
+
+import static io.airlift.configuration.ConfigBinder.configBinder;
+import static java.util.Objects.requireNonNull;
+
+public class ${classPrefix}Module
+        implements Module
```

Comment:
shouldn't this extend `AbstractConfigurationAwareModule`?

Reply:
Maybe, what does it provide? There's no comment on this class and I'm not that familiar with binders.

Reply:
AbstractConfigurationAwareModule allows you to bind config classes and use them for things like `conditionalModule` or any behaviour based on config. It's the better default than a plain `Module`.

Either would work though.